docs/tutorials/intro-to-rust.md

```diff
@@ -0,0 +1,418 @@
+</blockquote>
+
+You may add as many tests as you need following the pattern in this file. Similar to unit tests in other languages and frameworks, just add the attribute:
+
+`#[test]`
+
+above the block of code to have it run in the suite of tests.
+
+</hr>
+
+## Finally: test, compile, and deploy 🚀
+
+We're going to need two things to deploy this contract.
+
+1. a NEAR account, [created with Wallet](https://wallet.nearprotocol.com/create)
+2. `near-shell` installed [according to these instructions](/docs/development/near-clitool)
+
+Please use the links above if you haven't already, as we'll need those for deploying the smart contract.
+
+### Test the code
+
+```rust
+cargo test --package rust-counter-tutorial -- --nocapture
+```
+
+### Compile the code
+
+```bash
+RUSTFLAGS='-C link-arg=-s' cargo build --target wasm32-unknown-unknown --release
+```
+
+**Windows users**: please modify the above command as:
+```bash
+set RUSTFLAGS=-C link-arg=-s
+cargo build --target wasm32-unknown-unknown --release
+```
+
+The above command is essentially setting special flags and optimizing the resulting `.wasm` file. At the end of the day it's simply a customized `cargo build --release` command that should look familiar [from Chapter 1](https://doc.rust-lang.org/1.30.0/book/2018-edition/ch01-03-hello-cargo.html).
+
+### Login with near-shell
+
+We're going to use `near-shell` to login to our account created earlier at the Wallet site. In your command prompt, navigate to the directory containing the `Cargo.toml` file. (It also contains the `src` directory.)
+
+```bash
+near login
+```
+
+A link will be output after this command is run. Copy/paste the link into a browser. (Or on OS X, hold Command and click the link if your Terminal application allows it.)
+
+Follow the instructions in Wallet to authenticate your account, then head back to Terminal to complete the final step confirming the account name.
+
+Notice that our project directory now has a few new items:
+
+```bash
+.
+├── Cargo.lock  ⟵ created during build to lock dependencies
+├── Cargo.toml
+├── neardev     ⟵ created by "near login" containing keys
+├── src
+│  └── lib.rs
+└── target      ⟵ created during build, holds the compiled wasm
+```
+
+Now that we have the `neardev` folder with our keys, we can deploy the compiled contract to NEAR.
+
+In the following steps, please replace `YOUR_ACCOUNT_HERE` with the name of the account created in the NEAR Wallet.
+
+### Deploying
+
+```rust
+near deploy --wasmFile target/wasm32-unknown-unknown/release/rust_counter_tutorial.wasm --accountId YOUR_ACCOUNT_HERE
+```
+
+### Invoking
+
+We'll use `near-shell` to invoke methods on our smart contract.
+
+Increment:
+```bash
+near call YOUR_ACCOUNT_HERE increment --accountId YOUR_ACCOUNT_HERE
```

Comment:
I suggest we clarify the fact that we just "re-use" the account we have the contract deployed to do the contract call. I mean, this command can make the impression that you always have to pass the same contract id twice.

Reply:
great thought, Frol, thanks. I'll be pushing something up soon addressing this and a few other comments.

Reply:
The explanation is :100: !